libgcc/config/riscv/multi3.S

```diff
@@ -79,3 +80,4 @@ __multi3:
   mv  a0, t2
   mv  a1, t4
   jr  t0
+#endif
```

Comment:
I've rewritten a multi3.S to a C version from libgcc2.c, it's also improve muldi3 for RV32 and multi3 for RV64.